### DISCUSSION TITLE
Cant connect with real time using N8N SSE node - v0.22.20

### DISCUSSION THREAD

**Discussion by leogomide** [pocketbase/pocketbase#5445] [Q&A] [answered]
Not sure if its a issue or im missing something, so i will post in Q&A session 😅

# Prerequisites

### Localhost
Install latest n8n:

docker volume create n8n_data
docker run -it --rm --name n8n -p 5678:5678 -v n8n_data:/home/node/.n8n docker.n8n.io/n8nio/n8n

run pocketbase 0.22.20

### VPS

install easypanel
create n8n service
create pocketbase service

## Steps to Reproduce

create a new generic table 'cars', with fields 'brandName' and 'model'
apply api rules to everyone can see the records

# Expected Behavior

when adding or editing a 'cars' records, it should display some data in n8n SSE node

# Current Behavior

No data appears in n8n when some change are made to the 'cars' table

I have tested n8n node in a vps with sse.dev and it works fine

Tested pocketbase in vps with [ this  link,](https://svelte.dev/repl/2259e33e0661432794c0da05ad27e21d?version=3.47.0) replacing the url, nothing have show too

Also tested pocketbase and n8n in the vps using easypanel, dont work too

**Comment by ganigeorgiev** [comment]
I'm not familiar with n8n but the PocketBase realtime APIs require users to send a "subscribe" call with the specific topic for which they want to receive events (the `clientId` parameter is the one received from the initial `PB_CONNECT` connect message).

For more details please refer to https://pocketbase.io/docs/api-realtime/.

The easiest way to rule out PocketBase issue is to create a static `pb_public/index.html` file:

```html
<!DOCTYPE html />
<html>
<head>
    <meta charset="utf-8" />
    <title>Test</title>
</head>
<body>
    <script type="module">
        import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@master/dist/pocketbase.es.mjs"

        const pb = new PocketBase("http://127.0.0.1:8090");

        // authenticate if necessary
        await pb.collection("users").authWithPassword("test@example.com", "1234567890")

        // make sure that your user satisfies the List API rule of the "example" collection
        pb.collection("example").subscribe("*", (e) => {
            console.log("EVENT:", e);
        })
    </script>
</body>
</html>
```

Open the browser with its dev/inspect console and navigate to `http://127.0.0.1:8090`. On "example" collection record operation (create/update/delete) you should see a console log message with the action.

**Reply by ganigeorgiev** [reply, marked as answer]
I've also just noticed that in your Svelte link you are using the `EventSource.onmessage` prop but keep in mind that this prop (and its equivalent `message` event) is a special case and works only with "anonymous" messages (aka. without explicit `event` name field or event named "message"; all PocketBase messages are named). See the note at the bottom in https://developer.mozilla.org/en-US/docs/Web/API/EventSource.

I don't recommend connecting directly with `EventSource` if you are not sure what you are doing and instead prefer the JS SDK helpers as they also will handle realtime connection reuse, standardized auto reconnect+resubscribe across all browsers (Firefox and Chrome have different reconnect rules), etc.
